### DISCUSSION TITLE
taro H5 构建 js 目录除了 app.js 以外的 xx.js 如何配置不再生产（合并到app.js）里

### DISCUSSION THREAD

**Discussion by lishengzxc** [NervJS/taro#14924] [Q & A] [answered]
taro H5 构建 js 目录除了 app.js 以外的 xx.js 如何配置不再生产（合并到app.js）里 或者将 xx.js 的构建出的文件名固定？

![image](https://github.com/NervJS/taro/assets/12003270/86ee5060-86ee-4e11-9d82-bcff8b3e2553)
![image](https://github.com/NervJS/taro/assets/12003270/8b99df24-bfaf-4b14-a95b-56a8f12f151e)

看了下 855.js 里的内容 好像是 taro View Button Text... 这些组件的实现？

![image](https://github.com/NervJS/taro/assets/12003270/7f891af3-0d75-49f5-82f4-e20f5efe7c8c)

**Comment by ZakaryCode** [comment, marked as answer]
这些 chunk 是 web 端动态加载组件依赖生成的，通常来说可以通过 splitChunks 配置达成该需求，详情请查阅 webpack 相关文档；当然也可以通过[移除动态 import](https://nervjs.github.io/taro-docs/docs/babel-config#dynamic-import-node) 来实现（不建议）。

**Reply by lishengzxc** [reply]
搞定 谢谢
```
const config = {...chain.optimization.get('splitChunks')};
delete config.cacheGroups.taro;
chain.optimization.splitChunks(config)
```
